packages/sdk/src/AMM.ts

```diff
@@ -438,41 +438,53 @@ class AMM extends Base {
   }
 
   public async getDailyVolume () {
-    const provider = this.chain.provider
-    const block = await provider.getBlock('latest')
-    const endTimestamp = block.timestamp
-    if (!endTimestamp) {
-      throw new Error('getDailyVolume: could not retrieve timestamp')
-    }
-    const { totalVolume, totalVolumeFormatted } = await this.getAprForDay(endTimestamp)
+    const { volume, volumeFormatted } = await this.getYieldData()
     return {
-      volume: totalVolume,
-      volumeFormatted: totalVolumeFormatted
+      volume,
+      volumeFormatted
     }
   }
 
   public async getApr (days: number = 1) {
-    if (![1, 7, 30].includes(days)) {
-      throw new Error('invalid arg: valid days are: 1, 7, 30')
-    }
+    const { apr } = await this.getYieldData(days)
+    return apr
+  }
 
+  public async getApy (days: number = 1) {
+    const { apy } = await this.getYieldData(days)
+    return apy
+  }
+
+  public async getYieldData (days: number = 1) {
     const provider = this.chain.provider
     const block = await provider.getBlock('latest')
     const endTimestamp = block.timestamp
     const startTimestamp = DateTime.fromSeconds(endTimestamp)
       .minus({ days })
       .toSeconds()
 
-    const { totalFeesFormatted: feesEarnedToday, totalLiquidityFormatted: totalLiquidityToday } = await this.getAprForDay(endTimestamp)
+    const {
+      totalFeesFormatted: feesEarnedToday,
+      totalLiquidityFormatted: totalLiquidityToday,
+      totalVolume,
+      totalVolumeFormatted
+    } = await this.getAprForDay(endTimestamp)
 
     let feesEarnedDaysAgo = 0
     if (days > 1) {
       ;({ totalFeesFormatted: feesEarnedDaysAgo } = await this.getAprForDay(startTimestamp))
     }
+    const rate = (feesEarnedToday - feesEarnedDaysAgo) / totalLiquidityToday
```

Comment:
minor suggestion: move the calc logic to another function so that the worker can also use it for calculating staking apr (to dry calc code)

```suggestion
calcYield (interestEarned: number, interestEarnedAgo: number, principal: number, days: number) {
  const rate = (interestEarned - interestEarnedAgo) / principal
  const period = 365 / days
  const apr = rate * period
  const apy = (1 + rate) ** period - 1
  return { apr, apy }
}
```

```suggestion
    const { apr, apy } = this.calcYield(feesEarnedToday , feesEarnedDaysAgo, totalLiquidityToday, days)

    return {
      apr: Math.max(apr, 0),
      apy: Math.max(apy, 0),
      volume: totalVolume,
      volumeFormatted: totalVolumeFormatted
    }
```